### PR TITLE
INT-2812: Rewrite the react self-hosted docs

### PR DESCRIPTION
**Pull request by tiny-james**
Related Ticket: 

Description of Changes:

- Split react self-hosted docs into 4 (instead of 2)
- Rewrote and tested the whole process
- Reviewed and fixed the technical reference

Pre-checks:
- [x] Branch prefixed with `feature/6/` or `hotfix/6/`
- [x] `modules/ROOT/nav.adoc` has been updated (if applicable)
- [x] Files has been included where required (if applicable)
- [x] Files removed have been deleted, not just excluded from the build (if applicable)
- [x] (New product features only) Release Note added

Review:
- [x] Documentation Team Lead has reviewed
- [x] Product Manager has reviewed

### CODE REVIEW

File: modules/ROOT/nav.adoc [hunk 1]
```diff
@@ -16,8 +16,10 @@
 **** xref:rails-cloud.adoc[Ruby on Rails]
 *** Self-hosted
 **** React
-***** xref:react-pm.adoc[Using a package manager]
-***** xref:react-zip.adoc[Using a .zip package]
+***** xref:react-pm-host.adoc[Using a package manager with hosting]
+***** xref:react-pm-bundle.adoc[Using a package manager with bundling]
+***** xref:react-zip-host.adoc[Using a .zip package with hosting]
+***** xref:react-zip-bundle.adoc[Using a .zip package with bundling]
```
Comment (lnewson) [inline modules/ROOT/nav.adoc:22]: This reads a little strange but I don't really have any suggestions. The best I can think of is swapping the hosting/bundling to be first, e.g. `Hosting using a package manager` and `Bundling using a package manager`

**Review by lnewson — APPROVED** [review]

File: modules/ROOT/partials/integrations/react-quick-start.adoc [hunk 2]
```diff
@@ -23,22 +27,105 @@ npx create-react-app tinymce-react-demo
...
+  const log = () => {
+    if (editorRef.current) {
+      console.log(editorRef.current.getContent());
+    }
+  };
+  return (
+    <>
+      <Editor
+        apiKey='your-api-key'
+        onInit={(evt, editor) => editorRef.current = editor}
```
Comment (tiny-ben-tran) [inline modules/ROOT/partials/integrations/react-quick-start.adoc:56]:
```suggestion
        onInit={ (evt, editor) => editorRef.current = editor }
```
For better readability

File: modules/ROOT/nav.adoc (hunk 1 above)
Comment (tiny-james) [reply modules/ROOT/nav.adoc:22]: The problem is then it starts different to all the other headings that say just "Using a package manager" and "Using a .zip package".

File: modules/ROOT/partials/integrations/react-quick-start.adoc (hunk 2 above)
Comment (tiny-james) [reply modules/ROOT/partials/integrations/react-quick-start.adoc:56]: All react code style guides I could find disagree with this spacing.

Comment (tiny-james) [reply modules/ROOT/partials/integrations/react-quick-start.adoc:56]: For example this:
https://dev.to/abrahamlawson/react-style-guide-24pp
Has this example:
```jsx
function ItemList(props) {
  return (
    <ul>
      {props.items.map((item, index) => (
        <Item
          key={item.key}
          onClick={(event) => { doSomethingWith(event, item.name, index); }}
        />
      ))}
    </ul>
  );
}
```

**Review by tiny-ben-tran — APPROVED** [review]
On line 633 of `react-quick-start.adoc` file, I think the markup syntax is incorrect for `+http://localhost:<port>/my_react_application+`

**Comment by lnewson** [comment]
> On line 633 of `react-quick-start.adoc` file, I think the markup syntax is incorrect for `+http://localhost:<port>/my_react_application+`

This looks correct to me, as this will keep it as plain text instead of being transformed into a link.

**Review by metricjs — APPROVED** [review]

**Pull request merged**